backend/modules/metadata_store/prisma_store.py

```diff
@@ -64,32 +70,26 @@ async def aget_collection_by_name(
         self, collection_name: str, no_cache: bool = True
     ) -> Optional[Collection]:
         try:
-            collection: Optional[
-                "PrismaCollection"
-            ] = await self.db.collection.find_first(where={"name": collection_name})
-            if collection:
-                return Collection.model_validate(collection.model_dump())
-            return None
+            # Fetch the collection by name
+            collection: "PrismaCollection" = (
+                await self.db.collection.find_first_or_raise(
+                    where={"name": collection_name}
+                )
+            )
+            # Validate the collection and return it
+            return Collection.model_validate(collection.model_dump())
+        except RecordNotFoundError:
+            raise HTTPException(
+                status_code=404,
+                detail=f"Collection with name {collection_name!r} not found",
+            )
         except Exception as e:
             logger.exception(f"Failed to get collection by name: {e}")
             raise HTTPException(
                 status_code=500, detail="Failed to get collection by name"
             )
 
     async def acreate_collection(self, collection: CreateCollection) -> Collection:
-        try:
```

Comment:
No need to make a DB query and check if a collection exists before creating since Prisma will raise a `UniqueViolationError` upon duplicate record creation. Handled in line 103.